.clang-tidy

```diff
@@ -73,5 +73,3 @@ CheckOptions:
   value:           true
 
 HeaderFilterRegex: 'pybind11/.*h'
-
-WarningsAsErrors: '*'
```

Comment:
@henryiii Any reason this is removed? If there are explicit checks which are giving warnings, they can be negated as errors in this part of the list.